lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1201,18 +1215,22 @@ private synchronized void onConnected(ProtocolMessage message) {
                 // Add pending messages to the front of queued messages to be sent later
                 addPendingMessagesToQueuedMessages(false);
             } else {
+                System.out.println("resume_channel_test: resume has failed ");
                 // RTN15c3: resume failed
                 if (error != null){
                     Log.d(TAG, "connection resume failed with error: " + error.message);
                 }else { // This shouldn't happen but, putting it here for safety
                     Log.d(TAG, "connection resume failed without error" );
                 }
 
-                channels.reattachOnResumeFailure();
-                // Add any messages still pending from the previous transport (RTN19a) to the front of queued messages
-                // however, this time the pending messages have to have newly assigned `
-                // msgSerial`s. They can't simply be replayed, as they are in the successful resume case
+                //we are going to add pending messages and update pending queue state
                 addPendingMessagesToQueuedMessages(true);
+
+                //We are going to transfer those messages to channel level so that they are published after
+                //their respective channels are attached
+                channels.transferToChannels(new ArrayList<>(queuedMessages));
```

Comment:
I think we're going to have to change how we process unacknowledged `Message`s (as distinct from `PresenceMessage`s. According to the spec (https://sdk.ably.com/builds/ably/specification/main/features/#RTL6c2) all queued `Message`s (ie those that were queued to be sent, but not yet sent, in `ConnectionManager.queuedMessages`; plus those that were sent but are unacknowledged, in `ConnectionManager.pendingMessages`) should be re-sent immediately on establishment of the new connection.